compiler/lib/source_map.ml

```diff
@@ -520,7 +565,8 @@ module Standard = struct
   let to_file m file = Yojson.Raw.to_file file (json m)
 
   let invariant
-      { version; file = _; sourceroot = _; names; sources_content; sources; mappings } =
+      { version; file = _; sourceroot = _; names; sources_content; sources; mappings; _ }
```

Comment:
You could check that ignore list is a subset of sources, or add a comment if we should not do that